arangod/Aql/RegisterPlan.cpp

```diff
@@ -199,16 +216,41 @@ void RegisterPlanT<T>::increaseDepth() {
 }
 
 template <typename T>
-void RegisterPlanT<T>::addRegister() {
+RegisterId RegisterPlanT<T>::addRegister() {
   nrRegs[depth]++;
-  totalNrRegs++;
+  return totalNrRegs++;
+}
+
+template <typename T>
+void RegisterPlanT<T>::registerVariable(VariableId v, std::unordered_set<RegisterId>& unusedRegisters) {
+  RegisterId regId;
+
+  if (unusedRegisters.empty()) {
+    regId = totalNrRegs;
+    addRegister();
+  } else {
+    auto iter = unusedRegisters.begin();
```

Comment:
As this is an `unordered_set`, this line makes the register planning non-deterministic. I'd prefer it to make it deterministic, so maybe use a `set` instead?